dataflow-website/recipes/scaling/scdf-alert-webhook/src/main/java/io/spring/cloud/dataflow/alert/webhook/AlertWebHookApplication.java

```diff
@@ -0,0 +1,141 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.cloud.dataflow.alert.webhook;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.dataflow.rest.client.DataFlowOperations;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Elastic, Stream auto-scaling adapter for Spring Cloud DataFlow.
+ *
+ * The {@link AlertWebHookApplication} is configured as a Prometheus Alertmanager Webhook Receiver (https://github.com/prometheus/alertmanager),
+ * that listens for pre-configured Prometheus alerts and leverages the SCDF Scale API to scale out or in a preconfigured
+ * stream application.
+ *
+ * The {@link AlertWebHookApplication} listens for scale-out and scale-in alerts, configured by the
+ * {@link AlertWebHookProperties#getScaleOutAlertName} and the {@link AlertWebHookProperties#getScaleInAlertName}.
+ *
+ * On scale-out alert it increases the number of application instances to {@link AlertWebHookProperties#getScaleOutFactor} instances.
+ *
+ * On scale-in alert it decreases the number of application instances to {@link AlertWebHookProperties#getScaleInFactor} instances.
+ *
+ * The stream name should be provided either as a {@link AlertWebHookProperties#getScaleStreamName} property or an
+ * Alert label called: 'stream_name'
+ *
+ * The application name should be provided either as a {@link AlertWebHookProperties#getScaleApplicationName} property or an
+ * Alert label called: 'application_name'
+ *
+ * Internally the {@link AlertWebHookApplication} uses the Data Flow Scale REST API to scale the app instances in platform agnostic way.
+ * Use the spring.cloud.dataflow.client.server-uri property to configure url of the Data Flow server.
+ *
+ * @author Christian Tzolov
+ */
+
+@SpringBootApplication
+@RestController
+@EnableConfigurationProperties(AlertWebHookProperties.class)
+public class AlertWebHookApplication {
```

Comment:
I'm a bit confused why do we need a full-blown REST app to invoke the SCDF's endpoint? I had thought Prometheus had the webhook facility to invoke directly SCDF's `scale()` API, no?

Reply:
the boot application is the Prometheus webhook receiver.  The webhook interface has well defined interface and Alert Payloads in JSON body format. 
Prometheus doesn't know about the proprietary DataFlow apis. You need someone to convert the prometheus alerts into Data Flow compatible calls. And this is the role of the webhook.